internal/graph/graph.go

```diff
@@ -137,16 +137,23 @@ func (g *Graph) InsertConstructor(ctor interface{}) error {
 	return nil
 }
 
-// ValidateReturnTypes validates if ctor's return type is already insterted in the graph
-func (g *Graph) ValidateReturnTypes(ctype reflect.Type) error {
+// ValidateReturnTypes validates if Invoke func's return type is Provided to the graph
+// checkCachedObjects=true additionally checks if objects are resolved and cached in the graph
+func (g *Graph) ValidateReturnTypes(ctype reflect.Type, checkCachedObjects bool) error {
 	objMap := make(map[reflect.Type]bool, ctype.NumOut())
 	for i := 0; i < ctype.NumOut(); i++ {
 		objType := ctype.Out(i)
 		if _, ok := g.nodes[objType]; ok {
-			return errors.Wrapf(errRetNode, "ctor: %v, object type: %v", ctype, ctype.Out(i))
+			if checkCachedObjects {
```

Comment:
Added comment. This is needed when we Invoke already provided constructor. At that point, we just want to check if object is resolved and cached or not.